adapters/decenterads/decenterads.go

```diff
@@ -0,0 +1,155 @@
+package decenterads
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type DecenterAdsAdapter struct {
+	endpoint string
+}
+
+func (a *DecenterAdsAdapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	impressions := request.Imp
+	result := make([]*adapters.RequestData, 0, len(impressions))
+	errs := make([]error, 0, len(impressions))
+
+	for _, impression := range impressions {
+		if impression.Banner == nil && impression.Video == nil && impression.Native == nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "DecenterAds only supports banner, video or native ads",
+			})
+			continue
+		}
+		if len(impression.Ext) == 0 {
+			errs = append(errs, errors.New("impression extensions required"))
+			continue
+		}
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(impression.Ext, &bidderExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if len(bidderExt.Bidder) == 0 {
+			errs = append(errs, errors.New("bidder required"))
+			continue
+		}
+		var decenteradsExt openrtb_ext.ExtImpDecenterAds
+		err = json.Unmarshal(bidderExt.Bidder, &decenteradsExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if decenteradsExt.PlacementID == "" {
```

Comment:
The placement id length check isn't necessary. PBS-Core will enforce the json schema before reaching this point, where the placement id is already specified as required with a min length of 1.